src/libNode/FinalBlockProcessing.cpp

```diff
@@ -1364,20 +1371,20 @@ bool Node::ProcessMBnForwardTransaction(const bytes& message,
                           << " shard "
                           << entry.m_microBlock.GetHeader().GetShardId());
 
-    {
-      // skip for DS microblock submission
-      std::lock_guard<mutex> g(m_mediator.m_ds->m_mutexShards);
-      if (entry.m_microBlock.GetHeader().GetShardId() ==
-          m_mediator.m_ds->m_shards.size()) {
-        return true;
-      }
+    // skip soft confirmation for DSMB
+    if (isDSMB) {
+      return true;
     }
 
     // shard microblock only:
     // pre-process of early MBnForwardTxn submission
     // soft confirmation
     SoftConfirmForwardedTransactions(entry);
-    // [TODO] invoke txn distribution
+    // invoke txn distribution
+    if (!m_mediator.GetIsVacuousEpoch()) {
+      m_mediator.m_lookup->SendTxnPacketToShard(
```

Comment:
This sends the same txns which were sent when new dsepoch takes place.
so as soon as early confirmation is received for microblock of epoch 5, it will send the same txns fror epoch 6 which were already send after ds block were processed.
This causes no new txns being processed for epoch 6 by shards.